packages/elements/src/components/MosaicTableOfContents/types.ts

```diff
@@ -0,0 +1,43 @@
+export type TableOfContentsProps = {
+  tree: TableOfContentsItem[];
+  activeId: string;
+  Link: CustomLinkComponent;
+};
+
+export type CustomLinkComponent = React.ComponentType<{
+  className: string;
+  to: string;
+  children: React.ReactNode;
+}>;
+
+export type TableOfContentsItem = TableOfContentsDivider | TableOfContentsGroupItem;
+
+export type TableOfContentsDivider = {
+  title: string;
+};
+
+export type TableOfContentsGroupItem =
+  | TableOfContentsGroup
+  | TableOfContentsNodeGroup
+  | TableOfContentsNode
+  | TableOfContentsExternalLink;
+
+export type TableOfContentsGroup = {
+  title: string;
+  items: TableOfContentsGroupItem[];
+};
+
+export type TableOfContentsExternalLink = {
+  title: string;
+  url: string;
+};
+
+export type TableOfContentsNode<T = 'http_service' | 'http_operation' | 'model' | 'article' | 'overview'> = {
+  id: string;
+  slug: string;
+  title: string;
+  type: T;
+  meta: string;
```

Comment:
I don't really like this `meta` title.

It seems to be done in order to make the names more generic, but in `elements` it will be always literally just a method, like get, etc.? :thinking:

Reply:
In the DevPortal component, it could also be a version for models and APIs such as "v1" or "v2.1". Open to suggestions on renaming the prop name

Reply:
Hmm, then maybe it does make sense. Ok, let's keep it for now, we will see how it looks later with DevPortal.